src/server/ua_server.c

```diff
@@ -171,6 +171,24 @@ void UA_Server_delete(UA_Server *server) {
     }
     UA_Array_delete(server->namespaces, server->namespacesSize, &UA_TYPES[UA_TYPES_STRING]);
 
+#ifdef UA_ENABLE_FILETYPE_OBJECT_SUPPORT
+    UA_FileType *fileObject, *fileObject_tmp;
```

Comment:
Please keep this as much out of ua_server.c as possible.
Set up destructor set for the FileType object type.
So the cleanup happens automatically when the node is deleted.